src/dask_awkward/layers/layers.py

```diff
@@ -31,14 +31,29 @@ def mock(self):
 
     def __getstate__(self) -> dict:
         d = self.__dict__.copy()
-        try:
-            pickle.dumps(d["_meta"])
-        except (ValueError, TypeError, KeyError):
-            d.pop(
-                "_meta", None
-            )  # must be a typetracer, does not pickle and not needed on scheduler
+        if "_meta" in d:
+            try:
+                pickle.dumps(d["_meta"])
+            except (ValueError, TypeError, KeyError, NotImplementedError):
+                d[
+                    "_meta"
+                ] = (
+                    self._meta.layout.form
+                )  # must be a typetracer, does not pickle and not needed on scheduler
         return d
 
+    def __setstate__(self, state):
+        import awkward as ak
+
+        self.__dict__.update(state)
+        if "_meta" in state:
+            if hasattr(self._meta, "length_zero_array"):
+                self._meta = ak.Array(
+                    self._meta.length_zero_array(highlevel=False).to_typetracer(
+                        forget_length=True
+                    )
```

Comment:
this needs to also pass the array behavior to the final high level array that is being made.